### DISCUSSION TITLE
CSS Styles for use with Parallel Plugin

### DISCUSSION THREAD

**Discussion by ShaneSheppard** [swup/swup#810] [Help] [answered]
I can't seem to find a working example to get the swup parallel plugin to work with Webflow.
I am looking to recreate horizontal scroll transitions between pages.

Here is my code for swup which has no errors in the inspector - 
`const options = {
		containers: ["#swup"],
    plugins: [
    	new SwupParallelPlugin(),
        new SwupHeadPlugin()
    ],
 };
const swup = new Swup(options);`

I also have a script reload for webflow as follows - 
`function wfscriptreload() {
    window.Webflow && window.Webflow.destroy();
    window.Webflow && window.Webflow.ready();
    window.Webflow && window.Webflow.require( 'ix2' ).init();
    document.dispatchEvent( new Event( 'readystatechange' ) );
    }
  	document.addEventListener('swup:pageView', wfscriptreload);
    document.addEventListener("DOMContentLoaded", wfscriptreload);`
    
 I have followed this demo from the swup site as its perfect for my use - 
 https://replit.com/@swupjs/Swup-Demo-Slideshow

Can someone come back to me on getting this transition working as it seems to be an issue with JS not an HTML or CSS issue.

**Comment by hirasso** [comment]
Hi @ShaneSheppard I have never worked with webflow. You'll need to create a reduced test case somewhere online that we can look at, otherwise we can't help you.

**Comment by ShaneSheppard** [comment]
Hi @hirasso,

Here is a webflow test page i created - 
[swup-test-ss.webflow.io/tests/red-test](https://swup-test-ss.webflow.io/tests/red-test)

**Comment by hirasso** [comment, marked as answer]
@ShaneSheppard you are missing crucial styles for making it work. Have a look at the docs of [Parallel Plugin](https://swup.js.org/plugins/parallel-plugin/#styling). In your case, adding this to your styles fixes the transition:

```css
.page {
  display: grid;
  overflow: hidden;
  grid-template-areas: "main";
}
.page > .transition-slide {
  grid-area: main;
}
```

I also noticed that your script reload hook won't work with swup v4 ([docs](https://swup.js.org/hooks/)). You'll need to update it to this (colons instead of camelCase):

```diff
- document.addEventListener('swup:pageView', wfscriptreload)
+ document.addEventListener('swup:page:view', wfscriptreload)
```

**Reply by ShaneSheppard** [reply]
Thanks @hirasso,
Yes it looks like i skipped out on a few required styles at the beginning and got some of the Webflow reset wrong.
Thank you so much for your time and help.
